lib/rex/proto/http/http_logger_subscriber.rb

```diff
@@ -0,0 +1,49 @@
+# -*- coding: binary -*-
+
+require 'rex/socket'
+require 'rex/text'
+require 'digest'
+
+module Rex
+module Proto
+module Http
+
+class HttpLoggerSubscriber < HttpSubscriber
+  def initialize(logger:)
+    raise RuntimeError, "Incompatible logger" unless logger.respond_to?(:print_line) && logger.respond_to?(:datastore)
+    @logger = logger
+  end
+
```

Comment:
Yard docs:

```
  # (see Rex::Proto::Http::HttpSubscriber#on_request)
  def on_request(request)
```

```
  # (see Rex::Proto::Http::HttpSubscriber#on_response)
  def on_response(response)
```

Reply:
Not sure this is valid YARD. Should you not be using `@see`? See https://rubydoc.info/gems/yard/file/docs/Tags.md#see

Reply:
> Should you not be using @see here instead? See https://rubydoc.info/gems/yard/file/docs/Tags.md#see

I don't believe so:

> If a tag's data begins with (see OBJECT) it is considered a "reference tag". A reference tag literally copies the tag data by the given tag name from the specified OBJECT. For instance, a method may copy all [@param](https://rubydoc.info/gems/yard/file/docs/Tags.md#param) tags from a given object using the reference tag syntax

Example:

https://github.com/lsegal/yard/blob/698b22cc2340230856ab17a25cfefa889f1c8289/spec/docstring_spec.rb#L361-L362

Reply:
Huh interesting I wasn't aware of that, thanks for pointing that out @adfoster-r7!